File: backend/models/seller.py
"""
This module defines the Pydantic models used in the application.

It includes models for UploadImage,.
These models help in validating and serializing the data exchanged between the API and the database.
"""
from typing import List
from pydantic import BaseModel

class UploadImageRequset(BaseModel):
    img: str
    
class UploadImageResponse(BaseModel):
    """
    Model representing an image. 
    """
    img_id: str
    img_link: str
class UploadItemRequest(BaseModel):
    """
    Model representing upload agricultural_product request.
    """
    name: str
    price: str
    category: str
    total_quantity: str
    off_shelf_date: str
    img_link: str
    img_id: str
    seller_id: int
    unit: str
    location: str #location where item will be put

class ProductBasicInfo(BaseModel):
    id: int
    name: str
    upload_date: str
    off_shelf_date: str
class ProductInfo(BaseModel):
    id: int
    name: str
    price: int
    category: str
    total_quantity: int
    upload_date: str
    off_shelf_date: str
    img_link: str
    img_id: str
    unit: str

class ProductOrderInfo(BaseModel):
    order_id: int
    buyer_name: str
    quantity: int
    product_price: int
    status: str
    timestamp: str
    is_put: bool

class IsPutRequest(BaseModel):
    order_ids:List[int]

class UpdateOffShelfDateRequest(BaseModel):
    date: str

    

File: client/interfaces/tribe_resident/seller/seller.ts


export interface UploadItem {
    name: string
    price: string
    category: string
    total_quantity: string
    off_shelf_date: string
    img_link: string
    img_id: string
    seller_id: Number | null
  }
  export interface BasicProductInfo{
    id: Number
    name: string
    upload_date: string
    off_shelf_date: string
  }
  
  
  export interface CartItem {
    id: Number
    produce_id: Number
    name: string
    img_url: string
    price: Number
    quantity: Number
    seller_id: Number
  }


  export interface ProductInfo {
    id: Number
    name: string
    price: Number
    category: string
    total_quantity: Number
    upload_date: string
    off_shelf_date: string
    img_link: string
    img_id: string  
  }
  
  export interface ProductOrderInfo {
    order_id: Number
    buyer_name: string
    quantity: Number
    product_price: Number
    status: string
    timestamp: string
    is_put: boolean
  }
  
  export interface IsPutRequest {
    order_ids: Number[]
  }